tools/githelper.py

```diff
@@ -0,0 +1,129 @@
+import itertools
+import re
+import logging
+import subprocess
+import typing
+
+
+Line = typing.NamedTuple(
+    "Line",
+    [("sourcefile", str), ("number", int), ("text", str), ("added", bool)],
+)
+FileLines = typing.NamedTuple(
+    "FileLines",
+    [("filename", str), ("lines", typing.Sequence[typing.Tuple[int, int]])],
+)
+
+
+def get_toplevel_path() -> str:
+    logger = logging.getLogger(__name__)
+
+    cmd = ["git", "rev-parse", "--show-toplevel"]
+    logger.debug("Executing: %r", cmd)
+    return subprocess.check_output(cmd, text=True).strip()
+
+
+def get_changed_lines(
+    from_ref=None, filter_lines=None, include_files=None
+) -> typing.Iterable[Line]:
+    """Inspect `git diff-index` output, yields changed lines."""
+
+    logger = logging.getLogger(__name__)
+
+    cmd = [
+        "git",
+        "diff-index",
+        "--patch",
+        "--unified=0",
+        from_ref if from_ref else "HEAD",
+    ]
```

Comment:
Follow up of: https://github.com/mixxxdj/mixxx/pull/2981#discussion_r464683308

In case of merge commits we need to use here "git show" and use only ++lines containing resolved conflicts. 

In case of the Travis check, we need to use: 
```
git diff PRE_COMMIT_FROM_REF...PRE_COMMIT_TO_REF
```
to not include merge commits into the diff.